test/utils/FunctionCallGenerator.sol

```diff
@@ -129,7 +130,11 @@ abstract contract FunctionCallGenerator is InvariantFixtures {
             if (!isRegistered) {
                 revertData = _wrapCallFailedRevertData(IKeyManagement.KeyDoesNotExist.selector);
             }
-            return _revokeCall(keyHash, revertData);
+            // Cannot revoke the rootKeyHash since it cannot be registered 
+            else if (currentKeyHash == KeyLib.ROOT_KEY_HASH) {
+                revertData = _wrapCallFailedRevertData(IKeyManagement.KeyDoesNotExist.selector);
+            }
```

Comment:
The reason why we have to add this now is that the behavior introduced in this PR above (108-109) special case the root key hash, returning bytes32(0) when the selected key to revoke is the root key.

This is different than the existing behavior which just called `testKey.toKeyHash()`, using the underlying `KeyLib.hash()` function which does not special case the root key hash. Instead of returning bytes32(0) it would incorrectly calculate the hash based off of the root EOA's public key, which would _not_ be registered on the account and thus `isRegistered` would be false, so the revert would be caught above in lines 130-132

Reply:
Hm I see, but I think we can just do

if (!isRegistered || currentKeyHash == KeyLib.ROOT_KEY_HASH)

Reply:
Ah yeah sg